prov/verbs/src/fi_verbs.c

```diff
@@ -153,6 +153,73 @@ vrb_get_rdma_rai(const char *node, const char *service, uint64_t flags,
 	return ret;
 }
 
+static int vrb_get_sin_rai(const char *node, const char *service, uint64_t flags,
+		      const struct fi_info *hints, struct rdma_addrinfo **rai)
+{
+	struct sockaddr_ib *sib;
+	char gid[INET6_ADDRSTRLEN];
+	uint16_t ps;
+	uint16_t pkey;
+	uint8_t scope_id;
+	uint16_t port = 0;
+	int ret;
+
+	memset(gid, 0, sizeof(gid));
+	sib = calloc(1, sizeof(struct sockaddr_ib));
+
+	ret = sscanf(node,
+	             "[%64[^]]]" /* GID */ ":%" SCNx16 /* P_Key */
+	             ":%" SCNx16 /* port space */
+	             ":%" SCNx8 /* Scope ID */,
+	             gid,
+	             &pkey,
+	             &ps,
+	             &scope_id);
+
+	if ((ret != 4) || (inet_pton(AF_INET6, gid, sib->sib_addr.sib_raw) < 1))
+		return -FI_EINVAL;
+
+	sib->sib_family = AF_IB;
+	sib->sib_pkey = htons(pkey);
+	sib->sib_scope_id = htonll((uint64_t)scope_id);
+
+	if (service)
+		sscanf(service, "%"SCNu16, &port);
+
+	vrb_set_sid(ps, port, sib);
```

Comment:
The above code looks like it should be in something like ofi_str_to_sib(), which would be a common function to check and convert a string into sib.

Reply:
Thanks, I have changed the code accordingly.